app/components/slide_up_panel/slide_up_panel.js

```diff
@@ -0,0 +1,251 @@
+            }
+        });
+    };
+
+    updatePosition = (newPosition) => {
+        const {position} = this.state;
+        this.previousTop = newPosition;
+        position.setValue(newPosition);
+    };
+
+    renderDragIndicator = (containerPosition) => {
+        return (
+            <Animated.View
+                style={[containerPosition, styles.dragIndicatorContainer]}
+                {...this.panGesture.panHandlers}
+            >
+                <TouchableWithoutFeedback
+                    onPressIn={this.handlePressIn}
+                    onPressOut={this.handlePressOut}
+                >
+                    <View style={styles.dragIndicator}/>
+                </TouchableWithoutFeedback>
+            </Animated.View>
+        );
+    };
+
+    render() {
+        const {content, header, onRequestClose} = this.props;
+        const containerPosition = {
+            top: this.state.position,
+        };
+
+        return (
+            <View style={styles.viewport}>
+                <View
+                    ref={this.setBackdropRef}
+                    style={styles.backdrop}
+                    pointerEvents='box-only'
+                    onTouchEnd={() => {
+                        if (!this.isDragging) {
+                            onRequestClose();
+                        }
+                    }}
+                    {...this.panGesture.panHandlers}
+                />
+                {this.renderDragIndicator(containerPosition)}
+                <Animated.View
+                    ref={this.setViewRef}
+                    style={[containerPosition, styles.container]}
+                    {...this.panGesture.panHandlers}
+                >
+                    <TouchableWithoutFeedback
+                        onPressIn={this.handlePressIn}
+                        onPressOut={this.handlePressOut}
+                    >
+                        <View>
+                            <View style={styles.topHeader}/>
+                            {header}
+                        </View>
+                    </TouchableWithoutFeedback>
+                    {content}
+                </Animated.View>
+            </View>
+        );
+    }
+}
+
+const styles = StyleSheet.create({
+    viewport: {
+        flex: 1,
+    },
+    container: {
+        flex: 1,
+        backgroundColor: 'white',
+        borderTopRightRadius: 10,
+        borderTopLeftRadius: 10,
+    },
+    backdrop: {
+        backgroundColor: 'rgba(0, 0, 0, 0.3)',
+        position: 'absolute',
+        top: 0,
+        bottom: 0,
+        left: 0,
+        right: 0,
+    },
+    topHeader: {
+        height: TOP_HEADER_HEIGHT - 20,
+    },
+    dragIndicatorContainer: {
+        marginVertical: 10,
+        alignItems: 'center',
+        justifyContent: 'center',
+    },
+    dragIndicator: {
+        backgroundColor: 'white',
+        height: 5,
+        width: 62.5,
+        opacity: 0.9,
+        borderRadius: 25,
+    },
+});
```

Comment:
Yeah, I have an idea how to do unit test for this component. Not about the animation but the typical unit testing where doing similar to other components. Probably, I'll add it up in the coming days.